apps/rush-lib/src/cli/actions/VersionAction.ts

```diff
@@ -90,18 +90,7 @@ export class VersionAction extends BaseRushAction {
   }
 
   protected run(): Promise<void> {
-    // try to get the user email
-    const userEmail: string | undefined = GitPolicy.getUserEmail(this.rushConfiguration);
-
-    if (!userEmail) {
-      return Promise.reject(undefined);
-    }
-
-    if (!this._bypassPolicy.value) {
-      if (!GitPolicy.check(this.rushConfiguration, userEmail)) {
-        return Promise.reject(undefined);
-      }
-    }
+    const userEmail: string = GitPolicy.getUserEmail(this.rushConfiguration, this._bypassPolicy.value);
```

Comment:
>const [](start = 4, length = 5)

This async function is throwing a synchronous exception.  Please verify that all callers (synchronously) catch the exception.

Reply:
Alternatively, I use this idiom:

```javascript
function myAsyncOperation(): Promise<void> {
  return Promise.resolve().then(() => {
    // mixture of sync/async code goes here
  });
}
```


---
In reply to: [210127876](https://github.com/Microsoft/web-build-tools/pull/772#discussion_r210127876) [](ancestors = 210127876)